lib/fluent/plugin/elasticsearch_error_handler.rb

```diff
@@ -1,23 +1,26 @@
+require 'fluent/event'
 require_relative 'elasticsearch_constants'
 
 class Fluent::ElasticsearchErrorHandler
   include Fluent::ElasticsearchConstants
 
   attr_accessor :records, :bulk_message_count
-  class BulkIndexQueueFull < StandardError; end
-  class ElasticsearchOutOfMemory < StandardError; end
   class ElasticsearchVersionMismatch < StandardError; end
-  class UnrecognizedElasticsearchError < StandardError; end
   class ElasticsearchError < StandardError; end
+
   def initialize(plugin, records = 0, bulk_message_count = 0)
     @plugin = plugin
     @records = records
     @bulk_message_count = bulk_message_count
   end
 
-  def handle_error(response)
+  def handle_error(response, tag, records)
+    if records.length != response['items'].length
+      raise ElasticsearchError, "The number of records submitted do not match the number returned. Unable to process bulk response"
+    end
+    retry_stream = Fluent::MultiEventStream.new
```

Comment:
@jcantrill I replaced `retry_records` with `retry_stream` which is `MultiEventStream` instance.
I think that we can create an event stream for resubmitting within error handler.
What do you think?

Reply:
Seems reasonable